qsprpred/extra/data/descriptors/sets.py

```diff
@@ -194,35 +194,56 @@ class PaDEL(DescriptorSet):
         descriptors (list[str]): list of PaDEL descriptor names
     """
 
-    def __init__(self, descs: list[str] | None = None, ignore_3D: bool = True):
+    _notJSON = ["_nameMapping", "_padel", "_descriptors", *DescriptorSet._notJSON]
+
+    def __init__(
+        self,
+        descs: list[str] | None = None,
+        ignore_3d: bool = True,
+        n_jobs: int | None = None,
+    ):
         """Initialize a PaDEL calculator
 
         Args:
             descs: list of PaDEL descriptor short names
-            ignore_3D (bool): skip 3D descriptor calculation
+            ignore_3d (bool): skip 3D descriptor calculation
         """
         super().__init__()
+        self.nJobs = n_jobs or os.cpu_count()
         self._descs = descs
-        self._ignore3D = ignore_3D
+        self._ignore3D = ignore_3d
+        # Initialize name mapping
+        self._initMapping()
+        # Initialize descriptors and calculator
+        if descs is None:
+            self.descriptors = None
+        else:
+            self.descriptors = descs
+
+    @property
+    def supportsParallel(self) -> bool:
+        return False
+
+    def _initMapping(self):
         # Obtain default descriptor names
         self._nameMapping = {}
         for descriptor in PaDEL_descriptors:
             # Skip if desc is 3D and set to be ignored
-            if ignore_3D and descriptor.is_3D:
+            if self._ignore3D and descriptor.is_3D:
                 continue
             for name in descriptor.description.name:
                 self._nameMapping[name] = descriptor
-        # Initialize descriptors and calculator
-        if descs is None:
-            self.descriptors = None
-        else:
-            self.descriptors = descs
+
+    def __setstate__(self, state):
```

Comment:
We had a serialization issue here  with some members not being directly serializable anymore. This was fixed by leaving out some members  in  `_notJSON` and then re-initializing  them in `__setstate__`.